Improve documentation and examples

yadaiio opened the pull request:
This pull request fixes a few typos with the help of https://github.com/szepeviktor/typos-on-you. 

Builds on top of: https://github.com/clue/reactphp-sqlite/pull/43.

SimonFrings reviewed (approved):
Looks good to me :+1:

SimonFrings commented on README.md:
```diff
@@ -31,9 +31,9 @@ tar stream into the `Decoder` which emits "entry" events for each individual fil
 require __DIR__ . '/vendor/autoload.php';
 
-$stream = new ReadableResourceStream(fopen('archive.tar', 'r'));
+$stream = new React/Stream/ReadableResourceStream(fopen('archive.tar', 'r'));
```
You're currently using the wrong slashes in here, I added a suggested change below. My suggestions also aligns with PSR-4, maybe worth to take a look at https://www.php-fig.org/psr/psr-4/ 

```suggestion
$stream = new React\Stream\ReadableResourceStream(fopen('archive.tar', 'r'));
```

SimonFrings commented on README.md:
```diff
@@ -31,9 +31,9 @@ tar stream into the `Decoder` which emits "entry" events for each individual fil
+$stream = new React/Stream/ReadableResourceStream(fopen('archive.tar', 'r'));
 
-$decoder = new Decoder();
+$decoder = new Clue/React/Tar/Decoder();
```
You're currently using the wrong slashes in here, I added a suggested change below. My suggestions also aligns with PSR-4, maybe worth to take a look at https://www.php-fig.org/psr/psr-4/ 

```suggestion
$decoder = new Clue\React\Tar\Decoder();
```

SimonFrings reviewed (changes requested):
Thanks for updating this PR, added a few remarks below :+1:

These changes look similar to https://github.com/clue/reactphp-sqlite/pull/43, maybe include a reference to this ticket in your description and update the title?

yadaiio commented:
Thank you @SimonFrings for informing me about this. Changes should be okay by now, I also changed the title of this pull request and added the references. Thank you 👍

SimonFrings reviewed (approved):
Looks good to me :+1:

clue reviewed (approved):
@yadaiio Thanks for looking into this, changes LGTM! :shipit:
